docs/import/add_layer.js

```diff
@@ -1,32 +1,73 @@
+import {LayerType} from '../firebase_layers.js';
 import {convertEeObjectToPromise} from '../map_util.js';
+import {createLayerRow} from './add_disaster.js';
 import {getCurrentLayers, updateLayersInFirestore} from './add_disaster_util.js';
 
-export {processNewFeatureLayer};
-
-const layerIndex = 4;
+export {processNewEeLayer};
 
 /**
- * One-off function for processing a feature-collection-typed layer and putting
- * its color column info into firestore.
+ * One-off function for processing a new feature-collection-typed layer and
+ * putting its color column info into firestore.
+ * @param {string} asset ee asset path
+ * @param {enum} type LayerType
  * @return {Promise<void>} Finishes when the property information has been
  * written to firestore.
  */
-function processNewFeatureLayer() {
-  const layer = getCurrentLayers()[layerIndex];
-  const featureCollection = ee.FeatureCollection(layer['ee-name']);
-  const properties = featureCollection.first().propertyNames();
-  // TODO: check if there are over ~50 distinct values and don't do evaluate
-  // on the values if so.
-  const stats = properties.map((property) => {
-    const max = featureCollection.aggregate_max(property);
-    const min = featureCollection.aggregate_min(property);
-    const values = featureCollection.aggregate_array(property);
-    return ee.Dictionary.fromLists(
-        ['max', 'min', 'values'], [max, min, values]);
-  });
-  return convertEeObjectToPromise(ee.Dictionary.fromLists(properties, stats))
-      .then((columns) => {
-        layer['color-function']['columns'] = columns;
-        return updateLayersInFirestore();
+function processNewEeLayer(asset, type) {
+  switch (type) {
+    case LayerType.IMAGE:
+    case LayerType.IMAGE_COLLECTION:
+      const layer = {
+        'asset-type': type,
+        'ee-name': asset,
+        'display-name': '',
+        'display-on-load': false,
+      };
+      return prependToTable(layer);
+    case LayerType.FEATURE_COLLECTION:
+      const featureCollection = ee.FeatureCollection(asset);
+      const properties = featureCollection.first().propertyNames();
+      const stats = properties.map((property) => {
+        const max = featureCollection.aggregate_max(property);
+        const min = featureCollection.aggregate_min(property);
+        const values = ee.Algorithms.If(
+            ee.Number(featureCollection.aggregate_count_distinct(property))
+                .lte(ee.Number(25)),
+            featureCollection.aggregate_histogram(property), ee.Dictionary());
```

Comment:
Should we filter out columns that have only one unique value? Or at least not let them be selected for the discrete coloring scheme?

Reply:
If you try to add "path of the storm" feature collection, you get a whole bunch of columns, and no easy indication which is the one you should be using. Filtering out all the ones that are constant would make that a lot easier.